ui/components/app/edit-gas-popover/edit-gas-popover.component.js

```diff
@@ -16,25 +28,54 @@ import {
   updateTransaction,
 } from '../../../store/actions';
 
-export const EDIT_GAS_MODE = {
-  SPEED_UP: 'speed-up',
-  CANCEL: 'cancel',
-  MODIFY_IN_PLACE: 'modify-in-place',
-};
-
 export default function EditGasPopover({
-  popoverTitle,
-  confirmButtonText,
-  editGasDisplayProps,
+  popoverTitle = '',
+  confirmButtonText = '',
+  editGasDisplayProps = {},
+  defaultEstimateToUse = 'medium',
   transaction,
   mode,
   onClose,
 }) {
   const t = useContext(I18nContext);
   const dispatch = useDispatch();
   const showSidebar = useSelector((state) => state.appState.sidebar.isOpen);
+
+  const showEducationButton = mode === EDIT_GAS_MODES.MODIFY_IN_PLACE;
   const [showEducationContent, setShowEducationContent] = useState(false);
 
+  const [warning] = useState(null);
+
+  const [showAdvancedForm, setShowAdvancedForm] = useState(false);
+  const [
+    dappSuggestedGasFeeAcknowledged,
+    setDappSuggestedGasFeeAcknowledged,
+  ] = useState(false);
+
+  const {
+    maxPriorityFeePerGas,
+    setMaxPriorityFeePerGas,
+    maxPriorityFeePerGasFiat,
+    maxFeePerGas,
+    setMaxFeePerGas,
+    maxFeePerGasFiat,
+    estimatedMaximumNative,
+    isGasEstimatesLoading,
+    gasFeeEstimates,
+    gasEstimateType,
+    gasPrice,
+    setGasPrice,
+    gasLimit,
+    setGasLimit,
+    estimateToUse,
+    setEstimateToUse,
+    estimatedMinimumFiat,
+    estimatedMaximumFiat,
+    isMaxFeeError,
+    isMaxPriorityFeeError,
+    isGasTooLow,
+  } = useGasFeeInputs(defaultEstimateToUse);
```

Comment:
I don't have the proper context here, but passing this many props around seems like a red flag to me. This is quite difficult to understand. It was overly complex before too when it was isolated to the "edit-gas-display" component, and now it's worse because the complexity has found its way into a component API now as well. I don't see how we could make a component with over a dozen props easily understood or used.

A better way to handle UI complexity would be to track UI state in a Redux store, and grab it directly from Redux wherever a value is needed. Then derived state can be calculated using selectors. Then hopefully the UI logic would be just that - UI logic, not cluttered with the rules of how editing gas works.

Anyway - that's not something that was introduced here, nor can it be fixed here in this PR. Just commenting on it now that I've noticed it.

Reply:
Our goal is to get to a working solution in as iterative a process as possible. We've discussed moving much of this logic into redux but that will happen at a later point when we have a clearer picture of where we are heading.

Most values come directly from the background state, and are computed using conversion methods and other hooks (useCurrencyDisplay)

Other values come from the transaction being modified which we used to set in state and access in redux, or passed in through modal or popover.

So I'd like to take more time to think about how we integrate this into redux

Reply:
It's far easier to iterate with smaller and simpler APIs than it is with enormously complex ones. I don't think an iterative approach is in conflict with good API design.

When I suggested Redux I hadn't looked closely enough to be honest. I see now that some of this is input field state, and that definitely makes sense to be at the component level. So maybe some of this is best here, and maybe some is best as utility functions or selectors or as hooks broken out into more logical chunks. Or maybe some should be in Redux, I don't know. But I'm sure we can come up with a more readable abstraction than this enormous list of props.

Reply:
We know we can come up with something, as well. I don't disagree with your sentiment but we're doing the best we can with our current knowledge of the problem, in the time constraints we are under without breaking anything that is currently in place for transactions today (before the fork). The consolidation of logic into a hook was a purposeful step meant to get us into a place where abstracting that into a redux store (or wherever) would be easier in the future.